packages/models-library/src/models_library/clusters.py

```diff
@@ -222,29 +221,22 @@ class Cluster(BaseCluster):
         },
     )
 
-    @model_validator(mode="before")
-    @classmethod
-    def check_owner_has_access_rights(cls, values):
-        values = jsonable_encoder(values)
-
-        is_default_cluster = bool(values["id"] == DEFAULT_CLUSTER_ID)
-        owner_gid = values["owner"]
+    @model_validator(mode="after")
+    def check_owner_has_access_rights(self: Self) -> Self:
+        is_default_cluster = bool(self.id == DEFAULT_CLUSTER_ID)
+        owner_gid = self.owner
 
         # check owner is in the access rights, if not add it
-        access_rights = values.get("access_rights", values.get("accessRights", {}))
+        access_rights = self.access_rights.copy()
         if owner_gid not in access_rights:
             access_rights[owner_gid] = (
-                CLUSTER_USER_RIGHTS.model_dump()
-                if is_default_cluster
-                else CLUSTER_ADMIN_RIGHTS.model_dump()
+                CLUSTER_USER_RIGHTS if is_default_cluster else CLUSTER_ADMIN_RIGHTS
             )
         # check owner has the expected access
         if access_rights[owner_gid] != (
-            CLUSTER_USER_RIGHTS.model_dump()
-            if is_default_cluster
-            else CLUSTER_ADMIN_RIGHTS.model_dump()
+            CLUSTER_USER_RIGHTS if is_default_cluster else CLUSTER_ADMIN_RIGHTS
         ):
             msg = f"the cluster owner access rights are incorrectly set: {access_rights[owner_gid]}"
             raise ValueError(msg)
-        values["access_rights"] = access_rights
-        return values
+        object.__setattr__(self, "access_rights", access_rights)
```

Comment:
Add a reference to this trick to overcome "frozeness" this way we have a context of why this was done this way?

Reply:
ok